src/auth/authOS.ts

```diff
@@ -0,0 +1,246 @@
+  }
+}
+
+export const useAuthMethod = (storage: Storage, options?: UseQueryOptions<AuthMethodState, Error>) => {
+  const query = useQuery({
+    queryKey: ['authMethod'],
+    queryFn: () =>
+      Promise.resolve(authMethodKey)
+        .then(storage.getItem)
+        .then((storedAuthMethod) => (!isEmptyString(storedAuthMethod) ? storedAuthMethod : '""'))
+        .then(JSON.parse)
+        .then((parsedAuthMethod) => {
+          switch (parsedAuthMethod) {
+            case 'pin':
+              return {
+                isPIN: true,
+                isOS: false,
+                isNone: false,
+              } as const
+            case 'os':
+              return {
+                isPIN: false,
+                isOS: true,
+                isNone: false,
+              } as const
+            default:
+              return {
+                isPIN: false,
+                isOS: false,
+                isNone: true,
+              } as const
+          }
+        }),
+    ...options,
+  })
+
+  return {
+    authMethod: query.data,
+    ...query,
+  }
+}
+
+export const useSaveAuthMethod = (storage: Storage, options?: UseMutationOptions<void, Error, AuthMethod>) => {
+  const mutation = useMutationWithInvalidations({
+    mutationFn: (authMethod) => storage.setItem(authMethodKey, JSON.stringify(authMethod)),
+    invalidateQueries: [['authMethod']],
+    ...options,
+  })
+
+  return {
+    ...mutation,
+    saveAuthMethod: mutation.mutate,
+  }
+}
+
+export const useAuthOsAppKey = (storage: Storage, options?: UseQueryOptions<string, Error>) => {
+  const query = useQuery({
+    suspense: true,
+    queryKey: ['installationId'],
+    queryFn: () =>
+      storage
+        .getItem('/appSettings/installationId')
+        .then((data) => {
+          if (data != null) return data
+          throw new Error('Invalid state')
+        })
+        .then(JSON.parse),
+    ...options,
+  })
+
+  return query.data
+}
+
+export const useChangeAuthMethod = (storage: Storage, options?: UseMutationOptions<void, Error, AuthMethod>) => {
+  const mutation = useMutationWithInvalidations({
+    mutationFn: (authMethod) => storage.setItem(authMethodKey, JSON.stringify(authMethod)),
+    invalidateQueries: [['authMethod']],
+    ...options,
+  })
+
+  return mutation.mutate
+}
+
+export const useAuthOsErrorDecoder = () => {
+  const strings = useStrings()
+
+  const decoder = (error: Error | null | undefined) => {
+    if (!error || /code: 13/.test(error?.message)) return // 13 = Cancelled by user
+    if (/code: 7/.test(error?.message)) return strings.tooManyAttempts
+    return strings.unknownError
+  }
+
+  return decoder
+}
+
+export async function isOSAuthSupported() {
+  const checker = Platform.select({
+    android: () =>
+      Keychain.getSupportedBiometryType().then(
+        (supportedBioType) => supportedBioType === Keychain.BIOMETRY_TYPE.FINGERPRINT,
+      ),
+    ios: () =>
+      Promise.all([
+        Keychain.canImplyAuthentication({
```

Comment:
im not sure what this is
if the user has a biometry type, is that good enough?

Reply:
on `iOS` you can disable bio auth for that specific app